tidy3d/plugins/adjoint/components/data/data_array.py

```diff
@@ -312,28 +313,37 @@ def isel(self, **isel_kwargs) -> JaxDataArray:
 
         return self_sel
 
-    def sel(self, indexers: dict = None, method: str = "nearest", **sel_kwargs) -> JaxDataArray:
+    def sel(self, indexers: dict = None, method: str | None = None, **sel_kwargs) -> JaxDataArray:
```

Comment:
does this `|` fail in earlier versions of python? we actually recently stopped supporting 3.8 so maybe it's ok now

Reply:
Also, maybe we want it to be something like `method: typing.Literal[None, "nearest"]`?

Reply:
> does this | fail in earlier versions of python? we actually recently stopped supporting 3.8 so maybe it's ok now

Yes absolutely! Just checked, it was introduced in 3.10, so I'll have to change that. I'd _really_ like to use this & `match` statements though...

> Also, maybe we want it to be something like method: typing.Literal[None, "nearest"]?

Yes I was thinking about the `Literal` too, I'll add it. Not sure why I didn't go with it in the end. Probably it seemed redundant given the `if/else` check.

Reply:
my main concern was the docs, but actually I realized methods get documented using their docstrings (not the type annotations). We probably want to just add a proper `Parameters` and  `Returns` format in the docstring, mentioning `method`.  Could you add that while you're at it?